src/auth-service/bin/server.js

```diff
@@ -1,7 +1,165 @@
 const http = require("http");
-const app = require("../app");
+const express = require("express");
+const constants = require("@config/constants");
+const path = require("path");
+const cookieParser = require("cookie-parser");
+const app = express();
+const bodyParser = require("body-parser");
+const session = require("express-session");
+const MongoStore = require("connect-mongo")(session);
+const mongoose = require("mongoose");
+const { connectToMongoDB } = require("@config/database");
+connectToMongoDB();
+require("@config/firebase-admin");
+const morgan = require("morgan");
+const compression = require("compression");
+const helmet = require("helmet");
+const passport = require("passport");
+const isDev = process.env.NODE_ENV === "development";
+const isProd = process.env.NODE_ENV === "production";
+const options = { mongooseConnection: mongoose.connection };
+/** */
+const log4js = require("log4js");
+const debug = require("debug")("auth-service:server");
+const isEmpty = require("is-empty");
+const logger = log4js.getLogger(
+  `${constants.ENVIRONMENT} -- server start script`
+);
+const { logText, logObject } = require("@utils/log");
+
+// Express Middlewares
+app.use(
+  session({
+    secret: process.env.SESSION_SECRET,
+    store: new MongoStore(options),
+    resave: false,
+    saveUninitialized: false,
+  })
+); // session setup
+app.use(bodyParser.json({ limit: "50mb" })); // JSON body parser
+// Other common middlewares: morgan, cookieParser, passport, etc.
+if (isProd) {
+  app.use(compression());
+  app.use(helmet());
+}
+
+if (isDev) {
+  app.use(morgan("dev"));
+}
+
+app.use(passport.initialize());
+
+app.use(cookieParser());
+app.use(log4js.connectLogger(log4js.getLogger("http"), { level: "auto" }));
+app.use(express.json());
+app.use(
+  bodyParser.urlencoded({
+    extended: true,
+    limit: "50mb",
+    parameterLimit: 50000,
+  })
+);
+
+// Static file serving
+app.use(express.static(path.join(__dirname, "public")));
+
+app.use("/api/v1/users", require("@routes/v1"));
```

Comment:
the route definitions have been moved to other folders to keep this server start file neat and more focused. @BenjaminSsempala